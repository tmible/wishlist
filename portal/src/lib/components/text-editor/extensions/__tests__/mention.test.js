import { describe, expect, it, vi } from 'vitest';
import Mention from '../mention.js';

describe('text editor / extensions / mention', () => {
  it('should be defined', () => {
    expect(Mention).toMatchSnapshot();
  });

  describe('add attributes', () => {
    it('should add attributes', () => {
      expect(Mention.config.addAttributes()).toMatchSnapshot();
    });

    it('should parse href from HTML', () => {
      const getAttribute = vi.fn();
      Mention.config.addAttributes().href.parseHTML({ getAttribute });
      expect(getAttribute).toHaveBeenCalledWith('href');
    });
  });

  it('should parse HTML', () => {
    expect(Mention.config.parseHTML()).toMatchSnapshot();
  });

  it('should render HTML', () => {
    const bindValue = { options: { HTMLAttributes: { 'from-options': 'attribute' } } };
    const argument = { HTMLAttributes: { 'from-argument': 'attribute' } };
    expect(Mention.config.renderHTML.bind(bindValue)(argument)).toMatchSnapshot();
  });

  describe('add commands', () => {
    it('should add set command', () => {
      const setMark = vi.fn();
      Mention.config.addCommands().setMention('attributes')({ commands: { setMark } });
      expect(setMark).toHaveBeenCalledWith(Mention.name, 'attributes');
    });

    it('should add toggle command', () => {
      const toggleMark = vi.fn();
      Mention.config.addCommands().toggleMention('attributes')({ commands: { toggleMark } });
      expect(toggleMark).toHaveBeenCalledWith(Mention.name, 'attributes');
    });

    it('should add unset command', () => {
      const unsetMark = vi.fn();
      Mention.config.addCommands().unsetMention()({ commands: { unsetMark } });
      expect(unsetMark).toHaveBeenCalledWith(Mention.name);
    });
  });

  it('should add input rules', () => {
    expect(Mention.config.addInputRules()).toMatchSnapshot();
  });

  it('should add paste rules', () => {
    expect(Mention.config.addPasteRules()).toMatchSnapshot();
  });
});
