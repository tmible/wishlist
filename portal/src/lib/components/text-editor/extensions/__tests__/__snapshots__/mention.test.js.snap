// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text editor / extensions / mention > add attributes > should add attributes 1`] = `
{
  "class": {
    "default": "cursor-pointer",
  },
  "href": {
    "default": null,
    "parseHTML": [Function],
  },
  "rel": {
    "default": "noopener noreferrer nofollow",
  },
  "target": {
    "default": "_blank",
  },
}
`;

exports[`text editor / extensions / mention > should add input rules 1`] = `
[
  InputRule {
    "find": /\\(\\?:\\^\\|\\\\s\\)\\(\\(@\\[\\^\\\\s@\\]\\+\\)\\\\s\\)\\$/,
    "handler": [Function],
  },
]
`;

exports[`text editor / extensions / mention > should add paste rules 1`] = `
[
  PasteRule {
    "find": /\\(\\?:\\^\\|\\\\s\\)\\(\\(@\\[\\^\\\\s@\\]\\+\\)\\\\s\\)/g,
    "handler": [Function],
  },
]
`;

exports[`text editor / extensions / mention > should be defined 1`] = `
Mark {
  "child": null,
  "config": {
    "addAttributes": [Function],
    "addCommands": [Function],
    "addInputRules": [Function],
    "addPasteRules": [Function],
    "defaultOptions": {},
    "name": "mention",
    "parseHTML": [Function],
    "renderHTML": [Function],
  },
  "name": "mention",
  "options": {},
  "parent": null,
  "storage": {},
  "type": "mark",
}
`;

exports[`text editor / extensions / mention > should parse HTML 1`] = `
[
  {
    "tag": "a[data-entity="mention"]",
  },
]
`;

exports[`text editor / extensions / mention > should render HTML 1`] = `
[
  "a",
  {
    "from-argument": "attribute",
    "from-options": "attribute",
  },
  0,
]
`;
